library/components/TTooltip.vue

```diff
@@ -1,64 +1,122 @@
 <template>
-  <div
-    class="p-4 bg-[#1C1C21] text-white w-max h-max rounded absolute top-0 z-50"
-    :style="{ width, ...tooltipPositions }"
-  >
-    <slot></slot>
-    <div
-		class="bg-[#1C1C21] w-4 h-4 absolute rotate-45 m-auto"
-		:class="arrowPositions[position]"
-	></div>
-  </div>
+	<div @mouseover="handleMouseOver" @click="handleClick" ref="tooltipContainer">
+		<div ref="trigger">
+			<slot name="trigger">
+				<help-circle-outline-icon :size="12" />
+			</slot>
+		</div>
+		<div
+			class="p-4 bg-[#1C1C21] text-white w-max h-max rounded fixed z-50"
+			:style="{ width, ...positions.tooltip }"
+			ref="tooltip"
+			v-show="showTooltip"
+			@mouseleave="handleMouseLeave"
```

Comment:
I think that all the events should be on the trigger. As that is what is initiating the tooltip